### PR TITLE
fix(workers): Early return for rebaseWhen=never

### LINKED ISSUE #10358: Renovates updates MRs even when it says it won't
**How are you running Renovate?**

- [ ] WhiteSource Renovate hosted app on github.com
- [x] Self hosted

If using the hosted app, please skip to the next section.
Otherwise, if self-hosted, please complete the following:

Please select which platform you are using:

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server
- [ ] Gitea
- [ ] github.com
- [ ] GitHub Enterprise Server
- [ ] gitlab.com
- [x] GitLab self-hosted

Renovate version: 25.35.1-slim

**Describe the bug**

Somehow, renovate decides to rebase our MRs, while the `rebaseWhen` is set to never. It even mentions it in the MR descriptions: 

```
:recycle:️ **Rebasing**: Never, or you tick the rebase/retry checkbox.**
```

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Setting current branch to master","time":"2021-06-08T18:37:52.422Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","branchName":"master","latestCommitDate":"2021-05-26T15:05:44+00:00","msg":"latest commit","time":"2021-06-08T18:37:52.504Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getBranchPr(renovate/bomo-ical-bundle-2.x)","time":"2021-06-08T18:37:52.545Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"findPr(renovate/bomo-ical-bundle-2.x, undefined, open)","time":"2021-06-08T18:37:52.545Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getPr(944)","time":"2021-06-08T18:37:52.545Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getMR(944)","time":"2021-06-08T18:37:52.545Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"pr cannot be merged","time":"2021-06-08T18:37:53.367Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"branchExists=true","time":"2021-06-08T18:37:53.367Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"dependencyDashboardCheck=undefined","time":"2021-06-08T18:37:53.367Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"PR rebase requested=false","time":"2021-06-08T18:37:53.367Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Checking if PR has been edited","time":"2021-06-08T18:37:53.367Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Found existing branch PR","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Checking schedule(at any time, null)","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"No schedule defined","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","depName":"bomo/ical-bundle","daysElapsed":1,"stabilityDays":7,"msg":"Update has not passed stability days","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Branch already exists","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getBranchPr(renovate/bomo-ical-bundle-2.x)","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"findPr(renovate/bomo-ical-bundle-2.x, undefined, open)","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getPr(944)","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getMR(944)","time":"2021-06-08T18:37:53.386Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"pr cannot be merged","time":"2021-06-08T18:37:53.388Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Skipping stale branch check due to rebaseWhen=never","time":"2021-06-08T18:37:53.388Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"PR is conflicted","time":"2021-06-08T18:37:53.388Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Branch is not mergeable and needs rebasing","time":"2021-06-08T18:37:53.389Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Rebasing disabled by config","time":"2021-06-08T18:37:53.389Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Using reuseExistingBranch: true","time":"2021-06-08T18:37:53.389Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"manager.getUpdatedPackageFiles() reuseExistinbranch=true","time":"2021-06-08T18:37:53.389Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","msg":"Found composer lock file","time":"2021-06-08T18:37:53.409Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","depName":"bomo/ical-bundle","msg":"Rebasing branch after deps list has changed","time":"2021-06-08T18:37:53.421Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"manager.getUpdatedPackageFiles() reuseExistinbranch=false","time":"2021-06-08T18:37:53.421Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","depName":"bomo/ical-bundle","msg":"Starting search at index 774","time":"2021-06-08T18:37:53.441Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","depName":"bomo/ical-bundle","msg":"Found match at index 774","time":"2021-06-08T18:37:53.441Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","msg":"Found composer lock file","time":"2021-06-08T18:37:53.452Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFile":"composer.json","depName":"bomo/ical-bundle","msg":"Contents updated","time":"2021-06-08T18:37:53.461Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"composer.updateArtifacts(composer.json)","time":"2021-06-08T18:37:53.461Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Using composer cache /builds/4qstyzih/0/idb/renovate-runner/renovate/cache/others/composer","time":"2021-06-08T18:37:53.461Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Using composer constraint from config","time":"2021-06-08T18:37:53.466Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","cmd":"composer","args":"update bomo/ical-bundle --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction --no-scripts --no-autoloader","msg":"composer command","time":"2021-06-08T18:37:53.467Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Using docker to execute","time":"2021-06-08T18:37:53.467Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","depName":"docker.io/renovate/composer","scheme":"composer","constraint":"2.*","msg":"Found version constraint - checking for a compatible image to use","time":"2021-06-08T18:37:53.467Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","depName":"docker.io/renovate/composer","scheme":"composer","constraint":"2.*","version":"2.0.12","msg":"Found compatible image version","time":"2021-06-08T18:37:53.477Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","image":"docker.io/renovate/composer","tagConstraint":"2.*","tagVersioning":"composer","tag":"2.0.12","msg":"Resolved tag constraint","time":"2021-06-08T18:37:53.477Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Docker image is already prefetched: docker.io/renovate/composer:2.0.12","time":"2021-06-08T18:37:53.477Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","command":"docker run --rm --name=renovate_composer --label=renovate_child -v \"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\":\"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\" -v \"/builds/4qstyzih/0/idb/renovate-runner/renovate/cache\":\"/builds/4qstyzih/0/idb/renovate-runner/renovate/cache\" -e COMPOSER_CACHE_DIR -e COMPOSER_AUTH -w \"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\" docker.io/renovate/composer:2.0.12 bash -l -c \"composer update bomo/ical-bundle --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction --no-scripts --no-autoloader\"","msg":"Executing command","time":"2021-06-08T18:37:53.516Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","cmd":"docker run --rm --name=renovate_composer --label=renovate_child -v \"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\":\"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\" -v \"/builds/4qstyzih/0/idb/renovate-runner/renovate/cache\":\"/builds/4qstyzih/0/idb/renovate-runner/renovate/cache\" -e COMPOSER_CACHE_DIR -e COMPOSER_AUTH -w \"/builds/4qstyzih/0/idb/renovate-runner/renovate/repos/gitlab/idb/idb\" docker.io/renovate/composer:2.0.12 bash -l -c \"composer update bomo/ical-bundle --with-dependencies --ignore-platform-reqs --no-ansi --no-interaction --no-scripts --no-autoloader\"","durationMs":1369,"stdout":"","stderr":"Loading composer repositories with package information\nDependency \"symfony/framework-bundle\" is also a root requirement. Package has not been listed as an update argument, so keeping locked at old version. Use --with-all-dependencies (-W) to include root dependencies.\nUpdating dependencies\nLock file operations: 0 installs, 2 updates, 0 removals\n  - Upgrading bomo/ical-bundle (v1.0.14 => v2.0.1)\n  - Upgrading kigkonsult/icalcreator (v2.24.2 => v2.30.5)\nWriting lock file\nInstalling dependencies from lock file (including require-dev)\nPackage operations: 0 installs, 4 updates, 0 removals\n  - Downgrading symfony/mime (v5.3.0 => v4.4.25): Extracting archive\n  - Upgrading kigkonsult/icalcreator (v2.24.2 => v2.30.5): Extracting archive\n  - Upgrading bomo/ical-bundle (v1.0.14 => v2.0.1): Extracting archive\n  - Downgrading vich/uploader-bundle (1.18.0 => 1.17.1): Extracting archive\nPackage doctrine/doctrine-cache-bundle is abandoned, you should avoid using it. No replacement was suggested.\nPackage doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.\nPackage guzzlehttp/ringphp is abandoned, you should avoid using it. No replacement was suggested.\nPackage guzzlehttp/streams is abandoned, you should avoid using it. No replacement was suggested.\nPackage symfony/inflector is abandoned, you should avoid using it. Use use `EnglishInflector` from the String component instead instead.\n123 packages you are using are looking for funding.\nUse the `composer fund` command to find out more!\n","msg":"exec completed","time":"2021-06-08T18:37:54.846Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Returning updated composer.lock","time":"2021-06-08T18:37:54.907Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Commiting vendor files in vendor","time":"2021-06-08T18:37:54.907Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Updated 1 package files","time":"2021-06-08T18:37:54.908Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Getting updated lock files","time":"2021-06-08T18:37:54.908Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","packageFiles":["package.json"],"msg":"Writing package.json files","time":"2021-06-08T18:37:54.908Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Writing any updated package files","time":"2021-06-08T18:37:54.908Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","updatedArtifacts":["composer.lock"],"msg":"Updated 1 lock files","time":"2021-06-08T18:37:54.910Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Ensuring comment \"⚠️ Artifact update problem\" in #944 is removed","time":"2021-06-08T18:37:54.912Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Getting comments for #944","time":"2021-06-08T18:37:54.912Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Found 2 comments","time":"2021-06-08T18:37:56.371Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"2 file(s) to commit","time":"2021-06-08T18:37:56.372Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Committing files to branch renovate/bomo-ical-bundle-2.x","time":"2021-06-08T18:37:56.372Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","result":{"author":null,"branch":"renovate/bomo-ical-bundle-2.x","commit":"3abb3f6","root":false,"summary":{"changes":2,"insertions":41,"deletions":0}},"msg":"git commit","time":"2021-06-08T18:37:56.752Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","result":{"pushed":[],"branch":{"local":"renovate/bomo-ical-bundle-2.x","remote":"renovate/bomo-ical-bundle-2.x","remoteName":"origin"},"remoteMessages":{"all":["View merge request for renovate/bomo-ical-bundle-2.x:","https://git.snt.utwente.nl/idb/idb/-/merge_requests/944"]}},"msg":"git push","time":"2021-06-08T18:37:59.372Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":30,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","commitSha":"3abb3f6","msg":"Branch updated","time":"2021-06-08T18:37:59.979Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Got res with 0 results","time":"2021-06-08T18:38:00.081Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Updating renovate/stability-days status check state to yellow","time":"2021-06-08T18:38:00.081Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Checking if we can automerge branch","time":"2021-06-08T18:38:01.617Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"mergeStatus=no automerge","time":"2021-06-08T18:38:01.617Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Ensuring PR","time":"2021-06-08T18:38:01.617Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"There are 0 errors and 0 warnings","time":"2021-06-08T18:38:01.617Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getBranchPr(renovate/bomo-ical-bundle-2.x)","time":"2021-06-08T18:38:01.618Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"findPr(renovate/bomo-ical-bundle-2.x, undefined, open)","time":"2021-06-08T18:38:01.618Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getPr(944)","time":"2021-06-08T18:38:01.618Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"getMR(944)","time":"2021-06-08T18:38:01.618Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"pr cannot be merged","time":"2021-06-08T18:38:01.622Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Found existing PR","time":"2021-06-08T18:38:01.623Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Processing existing PR","time":"2021-06-08T18:38:01.894Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"Merge Request #944 does not need updating","time":"2021-06-08T18:38:01.894Z","v":0}
{"name":"renovate","hostname":"runner-4qstyzih-project-8198-concurrent-0","pid":13,"level":20,"logContext":"lHbmXWL6y","repository":"idb/idb","branch":"renovate/bomo-ical-bundle-2.x","msg":"PR is not configured for automerge","time":"2021-06-08T18:38:01.894Z","v":0}
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [ ] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [x] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Interestingly, I run renovate on another Gitlab instance where it does not constantly rebase all MRs. They share the exact same configuration.

Looking at the logs, it looks like the following check returns different results for the two environments, but I'm not sure why that is the case...

https://github.com/renovatebot/renovate/blob/76ae49c4e8cd1b7cbbab350ad336eea90536bd7f/lib/workers/branch/auto-replace.ts#L99

The master branch is updated before a rebase is triggered and it does indeed update the dependencies, but why would it trigger a rebase while the other gitlab environment doesn't? 🤔 Could it have something to do with caching between the runs?

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Early return from branch processing as soon as we have `rebaseWhen` set to `never`.

## Context:

Fixes #10358

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
Anything missing?

**Comment by zharinov** [comment]
I wonder if we need to force rebase via dependency dashboard or checkbox? How strict our `never` should be?

File: lib/workers/branch/index.ts [hunk 1]
```diff
@@ -391,6 +391,14 @@ export async function processBranch(
...
     }
+    if (config.rebaseWhen === 'never') {
+      logger.debug(`Skipping commit (rebaseWhen=never)`);
+      return {
+        branchExists,
+        prNo: branchPr?.number,
+        result: BranchResult.NoWork,
+      };
+    }
     config.forceCommit =
```
Comment (rarkins) [inline lib/workers/branch/index.ts:402]: Shouldn't this check be done *before* the rebaseWhen=never return? i.e.to allow manually requested rebase?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/workers/branch/index.ts (hunk 1 above)
Comment (zharinov) [reply lib/workers/branch/index.ts]: Now, `isConflicted` won't cause rebase while dashboard or PR checkboxes should

**Comment by viceice** [comment]
> I wonder if we need to force rebase via dependency dashboard or checkbox? How strict our `never` should be?

I think it should rebase when using checkbox / label / rename

File: lib/workers/branch/index.ts [hunk 2]
```diff
@@ -442,10 +448,7 @@ export async function processBranch(
...
         return { branchExists: false, result: BranchResult.Automerged };
       }
-      if (
-        mergeStatus === 'stale' &&
-        ['conflicted', 'never'].includes(config.rebaseWhen)
-      ) {
+      if (mergeStatus === 'stale' && config.rebaseWhen === 'conflicted') {
         logger.warn(
           'Branch cannot automerge because it is stale and rebaseWhen setting disallows rebasing - raising a PR instead'
         );
```
Comment (viceice) [inline lib/workers/branch/index.ts:454]: This looks wrong to me. 🤔

Comment (zharinov) [reply lib/workers/branch/index.ts]: Reverted this change. Or did you mean the whole condition?

**Review by viceice — APPROVED** [review]

File: lib/workers/branch/index.ts (hunk 2 above)
Comment (viceice) [reply lib/workers/branch/index.ts]: Yeah, the condition checks for conflict and the message says never. So they didn't match for me.

**Review by rarkins — APPROVED** [review]

**Pull request merged**